### PR TITLE
thread: Avoid crash with multiple Message-ID headers

### LINKED ISSUE #1911: SQL error if an email has two Message-ID headers
```
[INSERT INTO ss_ticket_email_info SET thread_id=61305, email_mid=Array, headers='Delivered-To: helpdesk@ra...[sanitized]...\nMessage-ID: \nTo: ***********@gmail.com\nFrom:\nSubject: Exactly why women will LOVE you tonight!\nMessage-ID: \nContent-Type: text/html;'] Unknown column 'Array' in 'field list'

---- Backtrace ----
#0 (root)/include/mysqli.php(177): osTicket->logDBError('DB Error #1054', '[INSERT INTO ss...')
#1 (root)/include/class.thread.php(782): db_query('INSERT INTO ss_...')
#2 (root)/include/class.thread.php(772): ThreadEntry->logEmailHeaders('61305', Array, 'Delivered-To: h...')
#3 (root)/include/class.thread.php(1127): ThreadEntry->saveEmailInfo(Array)
#4 (root)/include/class.thread.php(1138): ThreadEntry->create(Array)
#5 (root)/include/class.thread.php(1174): ThreadEntry->add(Array)
#6 (root)/include/class.thread.php(1154): Message->add(Array, Array)
#7 (root)/include/class.thread.php(168): Message->create(Array, Array)
#8 (root)/include/class.ticket.php(1677): Thread->addMessage(Array, Array)
#9 (root)/include/class.ticket.php(2754): Ticket->postMessage(Array, 'Email', false)
#10 (root)/include/api.tickets.php(130): Ticket::create(Array, Array, 'Email', true, true)
#11 (root)/include/api.tickets.php(161): TicketApiController->createTicket(Array)
#12 (root)/include/api.tickets.php(219): TicketApiController->processEmail()
#13 (root)/api/lpipe.php(44): RemotePipeApiController::process()
#14 {main}
```

### PR DESCRIPTION
**Pull request by greezybacon**
Maybe fixes #1911

### CODE REVIEW

File: include/class.thread.php [hunk 1]
```diff
@@ -803,10 +803,26 @@ function getFiles() {
...
 
-        if(!$vars || !$vars['mid'])
+        if (!$vars || !isset($vars['mid']))
             return 0;
 
-        $this->ht['email_mid'] = $vars['mid'];
+        // If the email contains more than one Message-ID header, use the
+        // last non-empty value
+        if (isset($vars['mid']) && is_array($vars['mid'])) {
+            foreach ($vars['mid'] as $M) {
```
Comment (protich) [inline include/class.thread.php]: Why foreach loop here?

Perhaps  `$mid= array_pop($vars['mid']); ` is sufficient.

Comment (greezybacon) [inline include/class.thread.php]: That won't verify if it is empty

Comment (protich) [inline include/class.thread.php]: Makes sense -- what about we filter empty IDs?
```git
diff --git a/include/class.mailparse.php b/include/class.mailparse.php
index 969cc56..67903e0 100644
--- a/include/class.mailparse.php
+++ b/include/class.mailparse.php
@@ -244,6 +244,9 @@ class Mail_Parse {
     function getMessageId(){
         if (!($mid = $this->struct->headers['message-id']))
             $mid = sprintf('<%s@local>', md5($this->getHeader()));
+        elseif (is_array($mid))
+            $mid = array_filter($mid);
+
         return $mid;
     }
 
```

Comment (greezybacon) [inline include/class.thread.php]: I think that is the most elegant solution. It will result in an empty array which equates with FALSE

**Comment by greezybacon** [comment]
Ok. I tried a more direct route: ensure the mail parsers figure out the array issue

**Pull request merged**